src/MongoDB.Driver/UpdateManyModel.cs

```diff
@@ -99,5 +100,15 @@ public override WriteModelType ModelType
         {
             get { return WriteModelType.UpdateMany; }
         }
+
+        /// <inheritdoc />
+        public override void ValidateAndThrowIfNotValid()
+        {
+            if (_update is PipelineUpdateDefinition<TDocument> && (_arrayFilters != null && _arrayFilters.Any()))
+            {
+                // We will never support arrayFilters in the pipeline-style update.
```

Comment:
This comment seems unnecessary. The error message already says essentially the same thing.